src/main/java/org/openstreetmap/atlas/geography/atlas/builder/proto/ProtoAtlasBuilder.java

```diff
@@ -0,0 +1,387 @@
+    }
+
+    /**
+     * Write an Atlas to a resource in the naive ProtoAtlas format.
+     *
+     * @param atlas
+     *            the Atlas to be written
+     * @param resource
+     *            the resource to write into
+     */
+    public void write(final Atlas atlas, final WritableResource resource)
+    {
+        final ProtoAtlasContainer.Builder protoAtlasBuilder = ProtoAtlasContainer.newBuilder();
+
+        // put the Atlas features into the ProtoAtlasBuilder
+        writePointsToBuilder(atlas, protoAtlasBuilder);
+        writeLinesToBuilder(atlas, protoAtlasBuilder);
+        writeAreasToBuilder(atlas, protoAtlasBuilder);
+        writeNodesToBuilder(atlas, protoAtlasBuilder);
+        writeEdgesToBuilder(atlas, protoAtlasBuilder);
+        writeRelationsToBuilder(atlas, protoAtlasBuilder);
+
+        final ProtoAtlasContainer protoAtlas = protoAtlasBuilder.build();
+        resource.writeAndClose(protoAtlas.toByteArray());
+    }
+
+    private void parseAreas(final PackedAtlasBuilder builder, final List<ProtoArea> areas)
+    {
+        final ProtoLocationConverter protoLocationConverter = new ProtoLocationConverter();
+        final ProtoTagListConverter protoTagListConverter = new ProtoTagListConverter();
+        areas.forEach(protoArea ->
+        {
+            final long identifier = protoArea.getId();
+            final List<Location> shapePoints = protoArea.getShapePointsList().stream()
+                    .map(protoLocationConverter::convert).collect(Collectors.toList());
+            final Polygon geometry = new Polygon(shapePoints);
+            final Map<String, String> tags = protoTagListConverter.convert(protoArea.getTagsList());
+            // TODO see Mike's PR note about duplicate points, since start and end are the same for
+            // polygons. Make sure this is not happening. I think it's fine since I am building the
+            // Polygon using the Polygon constructor from a list of points that contains no
+            // duplicates.
+            builder.addArea(identifier, geometry, tags);
+        });
+    }
+
+    private void parseEdges(final PackedAtlasBuilder builder, final List<ProtoEdge> edges)
+    {
+        final ProtoLocationConverter protoLocationConverter = new ProtoLocationConverter();
+        final ProtoTagListConverter protoTagListConverter = new ProtoTagListConverter();
+        edges.forEach(protoEdge ->
+        {
+            final long identifier = protoEdge.getId();
+            final List<Location> shapePoints = protoEdge.getShapePointsList().stream()
+                    .map(protoLocationConverter::convert).collect(Collectors.toList());
+            final PolyLine geometry = new PolyLine(shapePoints);
+            final Map<String, String> tags = protoTagListConverter.convert(protoEdge.getTagsList());
+            builder.addEdge(identifier, geometry, tags);
+        });
+    }
+
+    private void parseLines(final PackedAtlasBuilder builder, final List<ProtoLine> lines)
```

Comment:
Did you want to combine parseLines and parseAreas? The only difference is creating a PolyLine vs. a Polygon, which can be handled with a boolean flag.